test/framework/src/main/java/org/elasticsearch/common/util/concurrent/DeterministicTaskQueue.java

```diff
@@ -101,6 +101,17 @@ public void runAllTasksInTimeOrder() {
         }
     }
 
+    /**
+     * Run all {@code runnableTasks} and {@code deferredTasks} that are scheduled to run before or on the given {@code timeInMillis}.
+     */
+    public void runTasksUpToTimeInOrder(long timeInMillis) {
+        runAllRunnableTasks();
+        while (nextDeferredTaskExecutionTimeMillis <= timeInMillis) {
+            advanceTime();
```

Comment:
This might be slightly trappy: if there is no task scheduled at `timeInMillis` then this method returns leaving the current time earlier than requested. Should we advance `currentTime` to `timeInMillis` in that case?

Reply:
Yeah I think you are right. It feels more natural the current time is at the specified time once the method runs. I updated accordingly.